### PR TITLE
Fresdesk app to demonstrate request to FD api

### PR DESCRIPTION
**Pull request by shravan-balasubramanian**
#56

### CODE REVIEW

File: Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * @param {*} client 
+ * @param {*} iparams 
+ * @param {*} loggedInUser 
+ */
+function getAllAgentTickets(client, iparams, loggedInUser) {
+    return new Promise(function(resolve, reject) {
+            var { fd_domain, fd_api_key } = iparams,
+            options  = {
+                headers :  {
+                    Authorization : `Basic ${fd_api_key}`
```
Comment (ravirajthedeveloper) [inline Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js]: Could you make it as a secure iparam and change it to secure templating to access the Freshdesk API key?

File: Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            window.fd_domain = fd_domain;
+            return getAllAgentTickets(client, {fd_domain, fd_api_key}, loggedInUser);
+        })
+        .then(function(allAgentTickets) {
+            generateList(allAgentTickets)
+        })
+        .catch(function() {
+            return alertFreshdesk('Error retrieving tickets')
+        })
+}
```
Comment (ravirajthedeveloper) [inline Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js]: Add an empty line at the EOF in this file and other files that are missing it.

File: Freshdesk/v2/simple-fd-req-app/.gitignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.fdk
```
Comment (ravirajthedeveloper) [inline Freshdesk/v2/simple-fd-req-app/.gitignore]: The .gitignore is added globally for the whole repo. It is not required for each app. Kindly remove it as it is redundant.

File: Freshdesk/v2/simple-fd-req-app/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    .
+    ├── README.md                  This file
+    ├── app                        Contains the files that are required for the front end component of the app
+    │   ├── app.js                 JS to render the dynamic portions of the app
+    │   ├── icon.svg               Sidebar icon SVG file. Should have a resolution of 64x64px.
+    │   ├── freshdesk_logo.png     The Freshdesk logo that is displayed in the app
+    │   ├── style.css              Style sheet for the app
+    │   ├── template.html          Contains the HTML required for the app’s UI
+    ├── config                     Contains the installation parameters and OAuth configuration
+    │   ├── iparams.json           Contains the parameters that will be collected during installation
```
Comment (ravirajthedeveloper) [inline Freshdesk/v2/simple-fd-req-app/README.md]: The iparams.json file is missing in the PR. Kindly add it as it has been neglected by the .gitignore configuration in this app.

File: Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js (hunk 1 above)
Comment (shravan-balasubramanian) [reply Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js]: Addressed

File: Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js (hunk 2 above)
Comment (shravan-balasubramanian) [reply Freshdesk/v2/simple-fd-req-app/app/relatedTickets.js]: Addressed

File: Freshdesk/v2/simple-fd-req-app/.gitignore (hunk 3 above)
Comment (shravan-balasubramanian) [reply Freshdesk/v2/simple-fd-req-app/.gitignore]: Done

File: Freshdesk/v2/simple-fd-req-app/README.md (hunk 4 above)
Comment (shravan-balasubramanian) [reply Freshdesk/v2/simple-fd-req-app/README.md]: Done

**Pull request merged**